src/realm/sync/client.cpp

```diff
@@ -1982,40 +1999,39 @@ void SessionWrapper::handle_pending_client_reset_acknowledgement()
 {
     REALM_ASSERT(!m_finalized);
 
-    auto pending_reset = _impl::client_reset::has_pending_reset(*m_db->start_frozen());
-    REALM_ASSERT(pending_reset);
-    m_sess->logger.info("Tracking pending client reset of type \"%1\" from %2", pending_reset->type,
-                        pending_reset->time);
+    std::optional<PendingReset> pending_reset;
+    {
+        pending_reset = get_pending_reset_store()->has_pending_reset();
+        if (!pending_reset)
```

Comment:
Yes - because I removed the check for `pending_reset` in `ClientHistory::get_status()`, so it is unknown whether or not a pending reset exists before entering this function.